packages/squiggle-lang/src/rescript/Utility/E.res

```diff
@@ -572,12 +572,22 @@ module A = {
       |> (x => Ok(x))
     }
 
-  let getByOpen = (a, op, bin) =>
-    switch getBy(a, r => bin(op(r))) {
-    | Some(r) => Some(op(r))
-    | None => None
+  let getByWithFn = (a, fn, boolCondition) => {
```

Comment:
It took me longer to figure out the functionality than it needed to because the name wasn't direct. But that's because I had forgotten the context of what [`Belt.Array`](https://rescript-lang.org/docs/manual/latest/api/belt/array#getby) chooses to call this `getBy`. I'm a little more into `getByFmap`, even though you have performance reasons to do this imperative instead of a composition of `O.fmap` and `Array.getBy`, the semantics read into the name smoother if you evoke that it still can be understood as a composition of `fmap` and `getBy`

Reply:
Note, I assumed the functional approach was the performance problem at the level of some rescript internal before I looked back at the old `getByOpen` and realized it spuriously called `op(r)` twice, so it's no longer obvious to me that a composition of `O.fmap` and `Belt.Array.getBy` would have worse performance

Reply:
I think there's a case for type-hinting for the future developer, because we would like to know whether the application of `fn` is meant to happen before or after the predicate evaluation.

Reply:
Good point, I like getByFmap